File: routes/upbit.py
from flask import Blueprint, jsonify, request
from api.upbit.markets import get_markets
from api.upbit.candles import get_candles
from api.upbit.ticker import get_ticker

upbit_bp = Blueprint("upbit", __name__, url_prefix="/api")


@upbit_bp.route("/markets")
def markets():
    return jsonify(get_markets())


@upbit_bp.route("/candles")
def candles():
    market = request.args.get("market", "KRW-BTC")
    count = request.args.get("count", 90)
    return jsonify(get_candles(market, count))


@upbit_bp.route("/ticker")
def ticker():
    market = request.args.get("market", "KRW-BTC")
    return jsonify(get_ticker(market))

File: routes/upbit.js
const { getMarkets } = require("../api/upbit/markets");
const { getCandles } = require("../api/upbit/candles");
const { getTicker } = require("../api/upbit/ticker");

async function handleUpbitRoutes(req, res, pathname, query) {
  if (pathname === "/api/markets") {
    const data = await getMarkets();
    res.setHeader("Content-Type", "application/json");
    return res.end(JSON.stringify(data));
  }

  if (pathname === "/api/candles") {
    const data = await getCandles(query.market, query.count);
    res.setHeader("Content-Type", "application/json");
    return res.end(JSON.stringify(data));
  }

  if (pathname === "/api/ticker") {
    const data = await getTicker(query.market);
    res.setHeader("Content-Type", "application/json");
    return res.end(JSON.stringify(data));
  }

  return false;
}

module.exports = { handleUpbitRoutes };
